src/objective.jl

```diff
@@ -160,3 +160,36 @@ function objective_function(model::Model,
                    MOI.ObjectiveFunction{MOIFunType}())::MOIFunType
     return jump_function(model, func)
 end
+
+"""
+    set_objective_coefficient(model::Model, variable::VariableRef, coefficient::Real)
+
+Set the linear objective coefficient associated with `Variable` to `coefficient`.
+
+Note: this function will throw an error if a nonlinear objective is set.
+"""
+function set_objective_coefficient(model::Model, variable::VariableRef, coeff::Real)
+    if model.nlp_data !== nothing && model.nlp_data.nlobj !== nothing
```

Comment:
It's slightly cleaner to check `_nlp_objective_function(model) != nothing` instead of accessing the fields of `model`